### ISSUE TITLE
Keyword.merge/2 accepts invalid second argument if first argument is empty

### ISSUE THREAD

**Issue by amarandon** [elixir-lang/elixir#11673]
Hi, I just came across this behavior which got me scratching my head for a short moment while working on something:

iex(1)> Keyword.merge([a: 1], [{"b", 2}])
** (ArgumentError) expected a keyword list as the second argument, got: [{"b", 2}]
    (elixir 1.13.1) lib/keyword.ex:922: Keyword.merge/2
iex(1)> Keyword.merge([], [{"b", 2}])    
[{"b", 2}]

I would expect the second example to trigger the same error as the first one. Is the current behavior intended? I had a look at the code and it looks very clear so I'd be happy to try a PR if needed.

**Comment by josevalim** [comment]
Yes, checking for all arguments can be expensive. Other functions like get may not do it either.
